crates/transpiler/src/passes/disjoint_layout.rs

```diff
@@ -117,21 +120,25 @@ pub fn py_run_pass_over_connected_components(
         }
         Ok(run_func.call1((dag, py_cmap))?.unbind())
     };
-    match distribute_components(dag, target)? {
+    let components = {
+        let mut borrowed = dag.borrow_mut();
+        distribute_components(borrowed.deref_mut(), target)?
```

Comment:
strictly I think you can syntax-soup this as `&mut *borrowed` or something like that, but this is probably more legible

Reply:
lol, tbh I never would have though of trying `&mut *borrowed` that's a weird syntax. My strategy for these kind of things is to pull up the pyo3 docs on the pointer type and just search for `-> &mut docs` or whatever typing I'm looking for.